agent/consul/leader_test.go

```diff
@@ -756,20 +756,20 @@ func TestLeader_ReapTombstones(t *testing.T) {
 
 	// Make sure there's a tombstone.
 	state := s1.fsm.State()
-	func() {
+	retry.Run(t, func(r *retry.R) {
 		snap := state.Snapshot()
 		defer snap.Close()
 		stones, err := snap.Tombstones()
 		if err != nil {
-			t.Fatalf("err: %s", err)
+			r.Fatalf("err: %s", err)
```

Comment:
There are a great catch.

@freddygv did you ever write up a doc with "common mistakes that make tests flaky" I feel like you might have done that in the last round but I don't recall reading it.

It might be a nice to start compiling a checklist somewhere (eng docs for now?) of common reasons tests get flaky that just explain a few of the most common things in this PR. Something like:
 - Don't use timeouts or sleeps less than XXX milliseconds
 - Make sure you fatal `r` and not `t` in retry loops (wonder if we could build static analysis for that)
 - Use WaitForLeader instead of WaitForTestAgent in these cases...

And so on for all the wait for helpers and which times they are needed since some are subtle (like ones related to client sessions being created etc).

If you feel like you could make a start on that based on this work I think it would be great to capture while you have the context loaded in your mind and super useful for all of us to review the lessons you've learned.

Reply:
Sure thing. That's info is a bit scattered right now, there's some info in the contributing docs and like one other place.

I'll centralize it in the eng docs repo.